lib/Db/CategoryDbWrapper.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+namespace OCA\Cookbook\Db;
+
+use OCA\Cookbook\Entity\CategoryEntity;
+use OCA\Cookbook\Entity\impl\CategoryEntityImpl;
+use OCA\Cookbook\Entity\impl\CategoryMappingEntityImpl;
+use OCA\Cookbook\Entity\impl\RecipeEntityImpl;
+use OCP\IDBConnection;
+
+class CategoryDbWrapper extends AbstractDbWrapper {
+	private const CATEGORIES = 'cookbook_categories';
+	
+	/**
+	 * @var string
+	 */
+	private $userId;
+	
+	/**
+	 * @var IDBConnection
+	 */
+	private $db;
+	
+	public function __construct(string $UserId, IDBConnection $db) {
+		parent::__construct($db);
+		$this->userId = $UserId;
+		$this->db = $db;
+	}
+	
+	protected function fetchDatabase(): array {
+		$qb = $this->db->getQueryBuilder();
+		
+		$qb ->select('name')
+			->from(self::CATEGORIES)
+			->where('user_id = :uid')
+			->groupBy('name')
+			->orderBy('name');
+		
+		$qb->setParameter('uid', $this->userId);
+		
+		$res = $qb->execute();
+		
+		$ret = [];
+		while ($row = $res->fetch()) {
+			$entity = $this->createEntity();
+			$entity->setName($row['name']);
+			
+			$ret[] = $entity;
+		}
+		
+		$res->closeCursor();
+		
+		return $ret;
+	}
+	
+	/**
+	 * Store a single entity back to the database
+	 * @param CategoryEntity $category The entity to store
+	 */
+	public function store(CategoryEntity $category): void {
+		// We do not need to store anything here. The categories are just virtually generated.
```

Comment:
What about creating categories as actual objects that can have, e.g., a description or other metadata? 😇

Reply:
Simply put: No. :grin:

The current DB schema has only a category name stored in a mapping type table. I kept with that for now. See also #300.

I was planning to proceed as follows:
1. Create the entities and a DB abstraction layer parallel to the current main work.
2. Deprecate major parts of the current RecipeDbService and refactor the DbCacheService that use direct DB operations.
3. Migrate and refactor any current usage of any of these methods in the codebase
4. Migrate any pending PRs to the new system
5. Remove the direct DB access

After the DB access has been encapsulated as described here, we can add further setter/getter and fields in the DB without affecting the functionality of other parts.